src/app/store/reducers/mailboxes.reducers.ts

```diff
@@ -103,6 +106,19 @@ export function reducer(
       };
     }
 
+    case MailActionTypes.UPDATE_MAILBOX_ORDER: {
+      return { ...state, isUpdatingOrder: true };
+    }
+
+    case MailActionTypes.UPDATE_MAILBOX_ORDER_SUCCESS: {
+      return {
+        ...state,
+        mailboxes: action.payload.mailboxes,
+        currentMailbox: action.payload.mailboxes[0],
+        isUpdatingOrder: false,
```

Comment:
Create an action for failure as well so that `isUpdatingOrder` can be set to `false`

Reply:
yes, but that is minor, because there could be max internal server error from backend, which also should not happen for this api.